spring-pulsar/src/main/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtil.java

```diff
@@ -46,52 +43,46 @@ private ConsumerBuilderConfigurationUtil() {
 
 	/**
 	 * Configures the specified properties onto the specified builder in a manner that
-	 * avoids <a href="https://github.com/apache/pulsar/issues/11646">Pulsar issue</a>.
+	 * loads non-serializable properties. See
+	 * <a href="https://github.com/apache/pulsar/pull/18344">Pulsar PR</a>.
 	 * @param builder the builder
 	 * @param properties the properties to set on the builder
 	 * @param <T> the payload type
 	 */
-	@SuppressWarnings("unchecked")
 	public static <T> void loadConf(ConsumerBuilder<T> builder, Map<String, Object> properties) {
 
 		ConsumerConfigurationData<T> builderConf = ((ConsumerBuilderImpl<T>) builder).getConf();
 		Map<String, Object> propertiesCopy = new HashMap<>(properties);
 
-		// Remove and remember problem fields from input props and builder
-		DeadLetterPolicy deadLetterPolicy = getValueToApplyToBuilderAfterLoadConf(builderConf::getDeadLetterPolicy,
```

Comment:
Nice improvement @cbornet . Thank you for getting the fix in for the PulsarClient! Do you mind updating the [unit test](https://github.com/spring-projects-experimental/spring-pulsar/blob/bb12a7b666c59af178571eba47bb81e35732c6e1/spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtilTests.java#L84) to include the newly added fields?